### PR TITLE
Header hygiene: avoid `using namespace dd4hep` in headers; use explicit `dd4hep::` namespace for units

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
Through some header includes, we pick up a number of `using namespace dd4hep` instances, along with `#include "DD4hepUnits.h"`. This makes it very unclear which MeV is being used.

This PR removes all `using namespace dd4hep` instances in headers, and fixes the units to account for this.

In a few cases, the automatic replacements have picked up units in comment lines, but we can figure those out as code review suggestions.

### What kind of change does this PR introduce?
- [X] Bug fix (issue: unclear unit systems)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No.

### CODE REVIEW

File: src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc [hunk 1]
```diff
@@ -77,8 +77,8 @@ void CalorimeterClusterRecoCoG::AlgorithmProcess() {
       auto cl = reconstruct(pcl);
 
       if (m_log->level() <= spdlog::level::debug) {
-        //LOG_INFO(default_cout_logger) << cl.getNhits() << " hits: " << cl.getEnergy() / GeV << " GeV, (" << cl.getPosition().x / mm << ", " << cl.getPosition().y / mm << ", " << cl.getPosition().z / mm << ")" << LOG_END;
-        m_log->debug("{} hits: {} GeV, ({}, {}, {})", cl->getNhits(), cl->getEnergy() / GeV, cl->getPosition().x / mm, cl->getPosition().y / mm, cl->getPosition().z / mm);
+        //LOG_INFO(default_cout_logger) << cl.getNhits() << " hits: " << cl.getEnergy() / dd4hep::GeV << " dd4hep::GeV, (" << cl.getPosition().x / dd4hep::mm << ", " << cl.getPosition().y / dd4hep::mm << ", " << cl.getPosition().z / dd4hep::mm << ")" << LOG_END;
+        m_log->debug("{} hits: {} dd4hep::GeV, ({}, {}, {})", cl->getNhits(), cl->getEnergy() / dd4hep::GeV, cl->getPosition().x / dd4hep::mm, cl->getPosition().y / dd4hep::mm, cl->getPosition().z / dd4hep::mm);
```
Comment (wdconinc) [inline src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc:81]:
```suggestion
        //LOG_INFO(default_cout_logger) << cl.getNhits() << " hits: " << cl.getEnergy() / dd4hep::GeV << " GeV, (" << cl.getPosition().x / dd4hep::mm << ", " << cl.getPosition().y / dd4hep::mm << ", " << cl.getPosition().z / dd4hep::mm << ")" << LOG_END;
        m_log->debug("{} hits: {} GeV, ({}, {}, {})", cl->getNhits(), cl->getEnergy() / dd4hep::GeV, cl->getPosition().x / dd4hep::mm, cl->getPosition().y / dd4hep::mm, cl->getPosition().z / dd4hep::mm);
```

File: src/algorithms/calorimetry/CalorimeterIslandCluster.cc [hunk 2]
```diff
@@ -158,8 +158,8 @@ void CalorimeterIslandCluster::AlgorithmProcess()  {
...
         const auto& hit = hits[i];
-        //LOG_INFO(default_cout_logger) << fmt::format("hit {:d}: energy = {:.4f} MeV, local = ({:.4f}, {:.4f}) mm, global=({:.4f}, {:.4f}, {:.4f}) mm", i, hit->getEnergy() * 1000., hit->getLocal().x, hit->getLocal().y, hit->getPosition().x,  hit->getPosition().y, hit->getPosition().z) << LOG_END;
-        m_log->info("hit {:d}: energy = {:.4f} MeV, local = ({:.4f}, {:.4f}) mm, global=({:.4f}, {:.4f}, {:.4f}) mm", i, hit->getEnergy() * 1000., hit->getLocal().x, hit->getLocal().y, hit->getPosition().x,  hit->getPosition().y, hit->getPosition().z);
+        //LOG_INFO(default_cout_logger) << fmt::format("hit {:d}: energy = {:.4f} dd4hep::MeV, local = ({:.4f}, {:.4f}) dd4hep::mm, global=({:.4f}, {:.4f}, {:.4f}) dd4hep::mm", i, hit->getEnergy() * 1000., hit->getLocal().x, hit->getLocal().y, hit->getPosition().x,  hit->getPosition().y, hit->getPosition().z) << LOG_END;
+        m_log->info("hit {:d}: energy = {:.4f} dd4hep::MeV, local = ({:.4f}, {:.4f}) dd4hep::mm, global=({:.4f}, {:.4f}, {:.4f}) dd4hep::mm", i, hit->getEnergy() * 1000., hit->getLocal().x, hit->getLocal().y, hit->getPosition().x,  hit->getPosition().y, hit->getPosition().z);
```
Comment (wdconinc) [inline src/algorithms/calorimetry/CalorimeterIslandCluster.cc:162]:
```suggestion
        //LOG_INFO(default_cout_logger) << fmt::format("hit {:d}: energy = {:.4f} MeV, local = ({:.4f}, {:.4f}) mm, global=({:.4f}, {:.4f}, {:.4f}) mm", i, hit->getEnergy() * 1000., hit->getLocal().x, hit->getLocal().y, hit->getPosition().x,  hit->getPosition().y, hit->getPosition().z) << LOG_END;
        m_log->info("hit {:d}: energy = {:.4f} MeV, local = ({:.4f}, {:.4f}) mm, global=({:.4f}, {:.4f}, {:.4f}) mm", i, hit->getEnergy() * 1000., hit->getLocal().x, hit->getLocal().y, hit->getPosition().x,  hit->getPosition().y, hit->getPosition().z);
```

File: src/algorithms/calorimetry/ImagingClusterReco.h [hunk 3]
```diff
@@ -145,7 +145,7 @@ class ImagingClusterReco {
         // debug output
         if (m_log->level() == SPDLOG_LEVEL_DEBUG) {
             for (const auto &cl: clusters) {
-                m_log->debug( fmt::format("Cluster {:d}: Edep = {:.3f} MeV, Dir = ({:.3f}, {:.3f}) deg", cl->id(),
+                m_log->debug( fmt::format("Cluster {:d}: Edep = {:.3f} dd4hep::MeV, Dir = ({:.3f}, {:.3f}) deg", cl->id(),
```
Comment (wdconinc) [inline src/algorithms/calorimetry/ImagingClusterReco.h:148]:
```suggestion
                m_log->debug( fmt::format("Cluster {:d}: Edep = {:.3f} MeV, Dir = ({:.3f}, {:.3f}) deg", cl->id(),
```

File: src/algorithms/calorimetry/ImagingTopoCluster.h [hunk 4]
```diff
@@ -83,27 +83,27 @@ class ImagingTopoCluster {
...
+        layerDistEtaPhi[1] = u_layerDistEtaPhi[1] / dd4hep::rad;
+        sectorDist = m_sectorDist / dd4hep::mm;
+        minClusterHitEdep = m_minClusterHitEdep / dd4hep::GeV;
+        minClusterCenterEdep = m_minClusterCenterEdep / dd4hep::GeV;
+        minClusterEdep = m_minClusterEdep / dd4hep::GeV;
 
         // summarize the clustering parameters
         m_log->info( fmt::format("Local clustering (same sector and same layer): "
-                              "Local [x, y] distance between hits <= [{:.4f} mm, {:.4f} mm].",
+                              "Local [x, y] distance between hits <= [{:.4f} dd4hep::mm, {:.4f} dd4hep::mm].",
```
Comment (wdconinc) [inline src/algorithms/calorimetry/ImagingTopoCluster.h:98]:
```suggestion
                              "Local [x, y] distance between hits <= [{:.4f} mm, {:.4f} mm].",
```

File: src/algorithms/calorimetry/ImagingTopoCluster.h [hunk 5]
```diff
@@ -83,27 +83,27 @@ class ImagingTopoCluster {
...
                               localDistXY[0], localDistXY[1])
         );
         m_log->info( fmt::format("Neighbour layers clustering (same sector and layer id within +- {:d}: "
-                              "Global [eta, phi] distance between hits <= [{:.4f}, {:.4f} rad].",
+                              "Global [eta, phi] distance between hits <= [{:.4f}, {:.4f} dd4hep::rad].",
```
Comment (wdconinc) [inline src/algorithms/calorimetry/ImagingTopoCluster.h:102]:
```suggestion
                              "Global [eta, phi] distance between hits <= [{:.4f}, {:.4f} rad].",
```

File: src/algorithms/calorimetry/ImagingTopoCluster.h [hunk 6]
```diff
@@ -83,27 +83,27 @@ class ImagingTopoCluster {
...
                               m_neighbourLayersRange, layerDistEtaPhi[0], layerDistEtaPhi[1])
         );
         m_log->info( fmt::format("Neighbour sectors clustering (different sector): "
-                              "Global distance between hits <= {:.4f} mm.",
+                              "Global distance between hits <= {:.4f} dd4hep::mm.",
```
Comment (wdconinc) [inline src/algorithms/calorimetry/ImagingTopoCluster.h:106]:
```suggestion
                              "Global distance between hits <= {:.4f} mm.",
```

**Comment by DraTeots** [comment]
Does it close https://github.com/eic/EICrecon/issues/372 ?

**Comment by wdconinc** [comment]
> Does it close https://github.com/eic/EICrecon/issues/372 ? 

Maybe it does. I didn't run an exhaustive search and focused mainly on calorimetry.

**Comment by wdconinc** [comment]
I think the only remaining place is `src/algorithms/tracking/TrackParamTruthInit.cc` where you just have `using Acts::UnitConstants::mm` etc, but in a source file that's probably fine and doesn't need to get addressed here.

As part of another PR we also need to be more careful when reading from and writing to input and output collections. Right now we are just assuming that the DD4hep and Acts units are what we want in EDM4hep. That is only correct IF you have compiled DD4hep with `DD4HEP_USE_GEANT4_UNITS=OFF`.

**Review by DraTeots — APPROVED** [review]

**Pull request merged**